---
 cddl/compat/opensolaris/misc/fsshare.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/cddl/compat/opensolaris/misc/fsshare.c
+++ b/cddl/compat/opensolaris/misc/fsshare.c
@@ -34,7 +34,6 @@
 #include <fcntl.h>
 #include <fsshare.h>
 #include <libutil.h>
-#include <pathnames.h>	/* _PATH_MOUNTDPID */
 #include <signal.h>
 #include <stdio.h>
 #include <string.h>
@@ -47,6 +46,9 @@
 static void
 restart_mountd(void)
 {
+	fprintf (stderr, "warning: mountd isn't supported in "
+		"Debian GNU/kFreeBSD yet\n");
+#if 0
 	struct pidfh *pfh;
 	pid_t mountdpid;
 
@@ -62,6 +64,7 @@
 	}
 	/* We have mountd(8) PID in mountdpid varible. */
 	kill(mountdpid, SIGHUP);
+#endif
 }
 
 /*
