--- a/sys/cddl/compat/opensolaris/rpc/xdr.h
+++ b/sys/cddl/compat/opensolaris/rpc/xdr.h
@@ -60,10 +60,16 @@
 }
 
 #undef XDR_CONTROL
+#ifdef __GLIBC__
+#define	XDR_CONTROL(xdrs, req, op)					\
+	(xdrmem_control((xdrs), (req), (op)))
+
+#else
 #define	XDR_CONTROL(xdrs, req, op)					\
 	(((xdrs)->x_ops->x_control == NULL) ?				\
 	    xdrmem_control((xdrs), (req), (op)) :			\
 	    (*(xdrs)->x_ops->x_control)(xdrs, req, op))   
+#endif
 
 #endif	/* !_KERNEL */
 
