--- a/cddl/contrib/opensolaris/cmd/zpool/zpool_main.c
+++ b/cddl/contrib/opensolaris/cmd/zpool/zpool_main.c
@@ -5004,10 +5004,17 @@
 	if (cb.cb_poolname[0] != '\0') {
 		(void) printf(
 		    "If you boot from pool '%s', don't forget to update boot code.\n"
+#if defined(__FreeBSD__)
 		    "Assuming you use GPT partitioning and da0 is your boot disk\n"
 		    "the following command will do it:\n"
 		    "\n"
 		    "\tgpart bootcode -b /boot/pmbr -p /boot/gptzfsboot -i 1 da0\n\n",
+#else
+		    "Assuming you use GNU GRUB and da0 is your boot disk\n"
+		    "the following command will do it:\n"
+		    "\n"
+		    "\tgrub-install /dev/da0\n\n",
+#endif
 		    cb.cb_poolname);
 	}
 
--- a/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
+++ b/cddl/contrib/opensolaris/lib/libzfs/common/libzfs_pool.c
@@ -2718,11 +2718,18 @@
 			(void) fprintf(stderr, dgettext(TEXT_DOMAIN, "If "
 			    "you boot from pool '%s', you may need to update\n"
 			    "boot code on newly attached disk '%s'.\n\n"
+#ifdef __FreeBSD__
 			    "Assuming you use GPT partitioning and 'da0' is "
 			    "your new boot disk\n"
 			    "you may use the following command:\n\n"
 			    "\tgpart bootcode -b /boot/pmbr -p "
 			    "/boot/gptzfsboot -i 1 da0\n\n"),
+#else
+			    "Assuming you use GNU GRUB and 'da0' is "
+			    "your new boot disk\n"
+			    "you may use the following command:\n\n"
+			    "\tgrub-install /dev/da0\n\n"),
+#endif
 			    zhp->zpool_name, new_disk);
 		}
 		return (0);
