--- a/cddl/lib/libzfs/Makefile
+++ b/cddl/lib/libzfs/Makefile
@@ -8,7 +8,7 @@
 LIB=	zfs
 DPADD=	${LIBMD} ${LIBPTHREAD} ${LIBUMEM} ${LIBUTIL} ${LIBM} ${LIBNVPAIR} \
 	${LIBAVL} ${LIBZFS_CORE}
-LDADD=	-lmd -lpthread -lumem -lutil -lm -lnvpair -lavl -lzfs_core
+LDADD=	-lmd -lpthread -lumem -lutil -lavl -lzfs_core
 
 SRCS=	deviceid.c \
 	fsshare.c \
--- a/cddl/lib/libzpool/Makefile
+++ b/cddl/lib/libzpool/Makefile
@@ -57,7 +57,7 @@
 CFLAGS+=	-I${.CURDIR}/../../../lib/libthr/arch/${MACHINE_CPUARCH}/include
 
 DPADD=		${LIBMD} ${LIBPTHREAD} ${LIBZ}
-LDADD=		-lmd -lpthread -lz
+LDADD=		-lmd -lpthread -lz -lumem -lnvpair -lavl
 
 # atomic.S doesn't like profiling.
 NO_PROFILE=
