--- a/sys/cddl/compat/opensolaris/sys/param.h
+++ b/sys/cddl/compat/opensolaris/sys/param.h
@@ -27,11 +27,11 @@
  *
  */
 
+#include_next <sys/param.h>
+
 #ifndef _COMPAT_OPENSOLARIS_SYS_PARAM_H_
 #define _COMPAT_OPENSOLARIS_SYS_PARAM_H_
 
-#include_next <sys/param.h>
-
 #define	PAGESIZE	PAGE_SIZE
 
 #ifdef _KERNEL
--- a/cddl/compat/opensolaris/include/stdio.h
+++ b/cddl/compat/opensolaris/include/stdio.h
@@ -27,11 +27,11 @@
  *
  */
 
+#include_next <stdio.h>
+
 #ifndef _COMPAT_OPENSOLARIS_STDIO_H_
 #define _COMPAT_OPENSOLARIS_STDIO_H_
 
-#include_next <stdio.h>
-
 #define ftello64	ftello
 #define lseek64		lseek
 #define fseeko64	fseeko
--- a/cddl/compat/opensolaris/include/stdlib.h
+++ b/cddl/compat/opensolaris/include/stdlib.h
@@ -27,11 +27,11 @@
  *
  */
 
+#include_next <stdlib.h>
+
 #ifndef _COMPAT_OPENSOLARIS_STDLIB_H_
 #define _COMPAT_OPENSOLARIS_STDLIB_H_
 
-#include_next <stdlib.h>
-
 #define getexecname	getprogname
 
 #endif
--- a/cddl/compat/opensolaris/include/alloca.h
+++ b/cddl/compat/opensolaris/include/alloca.h
@@ -27,6 +27,10 @@
  *
  */
 
+#ifdef __GLIBC__
+#include_next <alloca.h>
+#endif
+
 #ifndef _COMPAT_OPENSOLARIS_ALLOCA_H_
 #define _COMPAT_OPENSOLARIS_ALLOCA_H_
 
